AWS opensearch docs (#4602)
SumoLogic/sumologic-documentation: docs/integrations/amazon-aws/amazon-opensearch-service.md

<img src={useBaseUrl('img/integrations/amazon-aws/amazon-opensearch-service-logo.png')} alt="Thumbnail icon" width="50"/>

> Amazon OpenSearch Service is a managed service that makes it easy to deploy, operate, and scale OpenSearch clusters in the AWS Cloud. An OpenSearch Service domain is synonymous with an OpenSearch cluster. Domains are clusters with the settings, instance types, instance counts, and storage resources that you specify. 

Reviewer: ```suggestion
Amazon OpenSearch Service is a managed service that makes it easy to deploy, operate, and scale OpenSearch clusters in the AWS Cloud. An OpenSearch Service domain is synonymous with an OpenSearch cluster, with the settings, instance types, instance counts, and storage resources that you specify. 
```

Author: This text is from AWS document which describes relation between domain and cluster. I think we can keep this as it is.

> The Sumo Logic app for Amazon OpenSearch collects CloudWatch logs, CloudWatch metrics and CloudTrail logs, provides a unified logs and metrics app that provides insights into the operations and utilization of your OpenSearch service. The preconfigured dashboards help you monitor the key metrics by domain names and nodes, view the OpenSearch events for activities, and help you plan the capacity of your OpenSearch service.

Reviewer: ```suggestion
- CloudWatch logs
- CloudWatch metrics 
- CloudTrail logs
```

> ## **Log and Metrics types[​](https://docs.aws.amazon.com/opensearch-service/latest/developerguide/monitoring.html)**

Reviewer: ```suggestion
```

The Sumo Logic app for Amazon OpenSearch uses:

* OpenSearch CloudWatch Logs. For details, see [here](https://docs.aws.amazon.com/opensearch-service/latest/developerguide/createdomain-configure-slow-logs.html).  
* OpenSearch CloudWatch Metrics. For details, see [here](https://docs.aws.amazon.com/opensearch-service/latest/developerguide/managedomains-cloudwatchmetrics.html).  
* OpenSearch using AWS CloudTrail. For details, see [here](https://docs.aws.amazon.com/opensearch-service/latest/developerguide/managedomains-cloudtrailauditing.html).

> ### **Sample cloudTrail log messages**

Reviewer: ```suggestion
### Sample CloudTrail log messages
```

```

> 

Reviewer: ```suggestion
```json

    },
    "accessPolicies": {...},
  "responseElements": {
    "domainStatus": {
      "created": true,
      "clusterConfig": {
        "zoneAwarenessEnabled": false,
        "instanceType": "m4.large.search",
        "dedicatedMasterEnabled": false,
        "instanceCount": 1
      },
      "cognitoOptions": {
        "enabled": false
      },
      "encryptionAtRestOptions": {
        "enabled": false
      },
      "advancedOptions": {
        "rest.action.multi.allow_explicit_index": "true"
      },
      "upgradeProcessing": false,
      "snapshotOptions": {
        "automatedSnapshotStartHour": 0
      },
      "eBSOptions": {
        "eBSEnabled": true,
        "volumeSize": 10,
        "volumeType": "gp2"
      },
      "engineVersion": "OpenSearch_1.0",
      "processing": true,
      "aRN": "arn:aws:es:us-west-1:123456789012:domain/test-domain",
      "domainId": "123456789012/test-domain",
      "deleted": false,
      "domainName": "test-domain",
      "accessPolicies": {...},
  "requestID": "12345678-1234-1234-1234-987654321098",
  "eventID": "87654321-4321-4321-4321-987654321098",
  "eventType": "AwsApiCall",
  "recipientAccountId": "123456789012"
}

> 

Reviewer: ```suggestion
```

```

> ### **Sample queries**

Reviewer: ```suggestion
### Sample queries
```

> Successful Events by Event Name

Reviewer: ```suggestion
```

> ```

Reviewer: ```suggestion
```sql title="Successful Events by Event Name"
```

> 

Reviewer: ```suggestion
```

account={{account}} region={{region}} namespace=aws/es "\"eventsource\":\"es.amazonaws.com\""
| json "userIdentity", "eventSource", "eventName", "awsRegion", "sourceIPAddress", "userAgent", "eventType", "recipientAccountId", "requestParameters", "responseElements", "requestID", "errorCode", "errorMessage" as userIdentity, event_source, event_name, region, src_ip, user_agent, event_type, recipient_account_id, requestParameters, responseElements, request_id, error_code, error_message nodrop
| where event_source = "es.amazonaws.com" 
| json field=userIdentity "accountId", "type", "arn", "userName"  as accountid, type, arn, username nodrop
| parse field=arn ":assumed-role/*" as user nodrop 
| parse field=arn "arn:aws:iam::*:*" as accountid, user nodrop
| json field=requestParameters "domainName" as domainname nodrop
| if (isBlank(accountid), recipient_account_id, accountid) as accountid
| where (tolowercase(domainname) matches tolowercase("{{domainname}}")) or isBlank(domainname)
| if (isEmpty(error_code), "Success", "Failure") as event_status
| if (isEmpty(username), user, username) as user
| count as event_count by event_name
| sort by event_count, event_name asc
```

> Write Latency by Domain Name (Metrics-based)

Reviewer: ```suggestion
```

> ```

Reviewer: ```suggestion
```sql title="Write Latency by Domain Name (Metrics-based)"
```

> 

Reviewer: ```suggestion
```

account={{account}} region={{region}} namespace=aws/es domainname={{domainname}} !nodeid=* metric=WriteLatency statistic = average | avg by domainname 
```

> ## **Collecting logs and metrics for the Amazon OpenSearch app**

Reviewer: ```suggestion
## Collecting logs and metrics for the Amazon OpenSearch app
```

> ### **Collecting Metrics for Amazon OpenSearch**

Reviewer: ```suggestion
### Collecting Metrics for Amazon OpenSearch
```

> 1. Configure a [Hosted Collector](https://help.sumologic.com/docs/send-data/hosted-collectors/configure-hosted-collector/).  

Reviewer: Can you please the steps from this document: https://help.sumologic.com/docs/integrations/amazon-aws/dynamodb/#collect-metrics-for-amazon-dynamodb

2. Configure an [Amazon CloudWatch Source for Metrics](https://help.sumologic.com/docs/send-data/hosted-collectors/amazon-aws/amazon-cloudwatch-source-metrics/) or [AWS Kinesis Firehose for Metrics Source](https://help.sumologic.com/docs/send-data/hosted-collectors/amazon-aws/aws-kinesis-firehose-metrics-source/) (Recommended).  
3. Namespaces. Select aws/es.  
4. Metadata. Add an account field to the source and assign it a value that is a friendly name/alias to your AWS account from which you are collecting metrics. The account field allows you to query metrics.  
   ![Metadata][image2]  
5. Click Save.

> ### **Collecting Amazon OpenSearch Events using CloudTrail**

Reviewer: ```suggestion
### Collecting Amazon OpenSearch CloudTrail Logs
```

> 1. Add an [AWS CloudTrail Source](https://help.sumologic.com/docs/send-data/hosted-collectors/amazon-aws/aws-cloudtrail-source/) to your Hosted Collector.  

Reviewer: Can you add the steps similar to the steps in this doc: https://help.sumologic.com/docs/integrations/amazon-aws/dynamodb/#collect-amazon-dynamodb-cloudtrail-logs

     ![Fields][image3]  
   * Access Key ID and Secret Access Key. Enter your Amazon [Access Key ID and Secret Access Key](http://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSGettingStartedGuide/AWSCredentials.html). Learn how to use Role-based access to AWS [here](https://help.sumologic.com/docs/send-data/hosted-collectors/amazon-aws/aws-sources/).  
   * Log File Discovery \-\> Scan Interval. Use the default of 5 minutes. Alternately, enter the frequency. Sumo Logic will scan your S3 bucket for new data. Learn how to configure Log File Discovery [here](https://help.sumologic.com/docs/send-data/hosted-collectors/amazon-aws/aws-sources/).  
   * Enable Timestamp Parsing. Select the Extract timestamp information from log file entries check box.  
   * Time Zone. Select Ignore time zone from the log file and instead use, and select UTC from the dropdown.  
   * Timestamp Format. Select Automatically detect the format.  
   * Enable Multiline Processing. Select the Detect messages spanning multiple lines check box, and select Infer Boundaries.  
2. Click Save.

> ### **Field in Field Schema**

Reviewer: ```suggestion
### Field in Field Schema
```

1. [Classic UI](https://help.sumologic.com/docs/get-started/sumo-logic-ui-classic/). In the main Sumo Logic menu, select Manage Data \> Logs \> Fields.  
   [New UI](https://help.sumologic.com/docs/get-started/sumo-logic-ui/). In the top menu select Configuration, and then under Logs select Fields. You can also click the Go To... menu at the top of the screen and select Fields.  
2. Search for the `"domainname"` field.  
3. If not present, create it. Learn how to create and manage fields [here](https://help.sumologic.com/docs/manage/fields/#manage-fields).

> ### **Field Extraction Rule(s)**

Reviewer: ```suggestion
### Field Extraction Rule(s)
```

Create a Field Extraction Rule for CloudTrail Logs. Learn how to create a Field Extraction Rule [here](https://help.sumologic.com/docs/manage/field-extractions/create-field-extraction-rule/).

```

Rule Name: AwsObservabilityESCloudTrailLogsFER
Applied at: Ingest Time
Scope (Specific Data): account=* eventname eventsource \"es.amazonaws.com\"
```

Parse Expression:

```

| json "userIdentity", "eventSource", "eventName", "awsRegion", "recipientAccountId", "requestParameters", "responseElements" as userIdentity, event_source, event_name, region, recipient_account_id, requestParameters, responseElements nodrop 
| where event_source = "es.amazonaws.com" 
| json field=userIdentity "accountId", "type", "arn", "userName"  as accountid, type, arn, username nodrop 
| parse field=arn ":assumed-role/*" as user nodrop 
| parse field=arn "arn:aws:iam::*:*" as accountid, user nodrop 
| json field=requestParameters "domainName" as domainname 
| if (isBlank(accountid), recipient_account_id, accountid) as accountid 
| toLowerCase(domainname) as domainname 
| "aws/es" as namespace 
| fields region, namespace, domainname, accountid
```

> ## **Centralized AWS CloudTrail Log Collection**

Reviewer: ```suggestion
## Centralized AWS CloudTrail Log Collection
```

In case, you have a centralized collection of CloudTrail logs and are ingesting them from all accounts into a single Sumo Logic CloudTrail log source, create the following Field Extraction Rule to map a proper AWS account(s) friendly name/alias. Create it if not already present or update it as required.

> 

Reviewer: ```suggestion

```
```

* Rule Name: AWS Accounts  
* Applied at: Ingest Time  
* Scope (Specific Data): `_sourceCategory=aws/observability/cloudtrail/logs`  

> * Parse Expression: Enter a parse expression to create an “account” field that maps to the alias you set for each sub account. For example, if you used the “dev” alias for an AWS account with ID "528560886094" and the “prod” alias for an AWS account with ID "567680881046", your parse expression would look like:

Reviewer: ```suggestion
```

#### Parse Expression

Enter a parse expression to create an “account” field that maps to the alias you set for each sub account. For example, if you used the `“dev”` alias for an AWS account with ID `"528560886094"` and the `“prod”` alias for an AWS account with ID `"567680881046"`, your parse expression would look like:
```

```

| json "recipientAccountId"
// Manually map your aws account id with the AWS account alias you setup earlier for individual child account
| "" as account
| if (recipientAccountId = "528560886094",  "dev", account) as account
| if (recipientAccountId = "567680881046",  "prod", account) as account
| fields account
```

> ## **Installing the Amazon OpenSearch app**

Reviewer: ```suggestion
## Installing the Amazon OpenSearch app
```

> Now that you have set up a collection for Amazon OpenSearch, install the Sumo Logic app to use the pre-configured searches and dashboards that provide visibility into your environment for real-time analysis of overall usage.

Reviewer: ```suggestion
Now that you have set up a collection for **Amazon OpenSearch**, install the Sumo Logic app to use the [pre-configured dashboards](#viewing-amazon-opensearch-dashboards) that provide visibility into your environment for real-time analysis of overall usage.

import AppInstall from '../../reuse/apps/app-install.md';

<AppInstall/>
```

> 

Reviewer: Delete from line 211 to 231.

     * Choose Enter a Custom Data Filter, and [enter a custom source category](https://help.sumologic.com/docs/get-started/apps-integrations/#custom-data-filters) beginning with an underscore. For example, `_sourceCategory=MyCategory`.  
   * Folder Name. You can retain the existing name or enter a custom name of your choice for the app.  
   * All Folders (optional). The default location is the Personal folder in your Library. If desired, you can choose a different location and/or click New Folder to add it to a new folder.  
5. Click Next.  
6. Look for the dialog confirming that your app was installed successfully.  
   ![app-success.png][image4]

Post-installation

Once your app is installed, it will appear in your Personal folder or the folder that you specified. From here, you can share it with other users in your organization. Dashboard panels will automatically start to fill with data matching the time range query received since you created the panel. Results won't be available immediately, but within about 20 minutes, you'll see completed graphs and maps.

## **Viewing Amazon OpenSearch dashboards**

> 

Reviewer: import ViewDashboards from '../../reuse/apps/view-dashboards.md';

<ViewDashboards/>

> ### **Amazon Opensearch \- Overview**

Reviewer: ```suggestion
### Amazon Opensearch - Overview
```

> The OpenSearch \- Overview dashboard provides a comprehensive view of the OpenSearch cluster's health, performance, and resource utilization. It offers real-time insights into cluster status, CPU and memory usage, storage metrics, document management, and read/write latencies across different domains.

Reviewer: ```suggestion
The **Amazon OpenSearch - Overview** dashboard provides a comprehensive view of the OpenSearch cluster's health, performance, and resource utilization. It offers real-time insights into cluster status, CPU and memory usage, storage metrics, document management, and read/write latencies across different domains.
```

Use this dashboard to:

* Monitor the overall health of OpenSearch clusters with color-coded status indicators (green, yellow, red) and quickly identify the number of clusters in each state.  

> * Track resource utilization, including average CPU and JVM memory usage, both overall and by individual domain names.  

Reviewer: ```suggestion
* Track resource utilization, including average CPU and JVM memory usage, by both overall and individual domain names.  
```
